### PR TITLE
add autocomplete and search for monarch

### PR DESCRIPTION
**Pull request by kshefchek**
- [x] Add lay person autocomplete
- [x] Add monarch autocomplete
- [x] Add monarch search
-  [x] fix lint warnings

### CODE REVIEW

**Comment by kshefchek** [comment]
I've implemented autocomplete and slightly adjusted the previous search API call:
- adds prefix filter, boost function
- returns count of docs
- longest highlight

**Comment by cmungall** [comment]
Looks good so far

Do we have a document that describes the various dependencies. What solr fields are expected, what are the expectations on how they are indexed, and how they get populated in the monarch pipeline

**Comment by kshefchek** [comment]
I've used MagicMock to mock the PySolr.search() method so that, GolrSearchQuery.search(), GolrSearchQuery.autocomplete(), and GolrLayPersonSearch.autocomplete() can be tested.

It looks like we needed to specify the marshmallow version in setup.py, will rebase some of these commits to clean it up.

File: ontobio/sparql/sparql_ontology.py [hunk 1]
```diff
@@ -122,9 +122,9 @@ def resolve_names(self, names, is_remote=False, synonyms=False, **args):
             results = set()
             for name in names:
                 results.update( self._search(name, 'rdfs:label', **args) )
-            if synonyms:
-                for pred in OIO_SYNS.values():
-                    results.update( self._search(name, pred, **args) )
+                if synonyms:
```
Comment (DoctorBud) [inline ontobio/sparql/sparql_ontology.py]: It appears that there may be TAB characters here instead of SPACEs. Please check.

**Comment by DoctorBud** [comment]
+1
looks good to me

**Pull request merged**